app/src/main/java/org/thoughtcrime/securesms/preferences/ShareLogsDialog.kt

```diff
@@ -35,83 +29,84 @@ import java.util.concurrent.TimeUnit
 
 class ShareLogsDialog : DialogFragment() {
 
-    private var shareJob: Job? = null
-
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog = createSessionDialog {
         title(R.string.dialog_share_logs_title)
         text(R.string.dialog_share_logs_explanation)
-        button(R.string.share) { shareLogs() }
+        button(R.string.share) {
+            shareLogs()
+        }
         cancelButton { dismiss() }
     }
 
     private fun shareLogs() {
-        shareJob?.cancel()
-        shareJob = lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
Was it still useful to schedule the work?

could move it down, so it is just around the actual IO, and grab a reference to `ApplicationContext` around here first.